Equipo2Hosteleria/src/main/java/Vista/PanelPedidos.java

```diff
@@ -187,59 +271,58 @@ public void actionPerformed(ActionEvent arg0) {
 		};
 	}
 	
-	private ActionListener listenerBotonPasarACaja(ControladorPanelPedidos controladorPanelPedidos) {
+	private ActionListener listenerBotonPagar(ControladorPanelPedidos controladorPanelPedidos) {
 		return new ActionListener() {
 			public void actionPerformed(ActionEvent arg0) {
 				if(ControlarCaja==0) {
 					JOptionPane.showMessageDialog(null, "A�ade productos.");
-				}else if (rdbtnEntregaADomicilio.isSelected() && !textPane_Direccion.getText().equals("")) {
+				}else if (rdbtnEntregaADomicilio.isSelected() && !textField_Direccion.getText().equals("")) {
```

Comment:
Estos IF lo único que tiene de diferencia es los parámetros que le pasas a cada tipo, podéis hacerlo estableciendo una variable recogida o entrega y llamando solamente una vez a ccionadoBottonPasarACajaPanelPedidos etc etc al salir del if

Reply:
revisadlo en la pantalla de ticket/factura

Reply:
Ya está optimizado en ambas pantallas.